### ISSUE TITLE
iris.exceptions.NotYetImplementedError: Coord AuxCoord

### ISSUE THREAD

**Issue by SarahAlidoost** [SciTools/iris#4000]
## 🐛 Bug Report
The order in which a mathematical operation is applied on a **'iris.coords.AuxCoord'** and **'iris.cube.Cube'** makes a difference, i.e.
`iris.cube.Cube` / `iris.coords.AuxCoord` works fine whereas `iris.coords.AuxCoord` / `iris.cube.Cube` returns the error as:

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/sarah/anaconda3/envs/iris/lib/python3.8/site-packages/iris/coords.py", line 464, in __truediv__
    return self.__binary_operator__(other, self._MODE_DIV)
  File "/home/sarah/anaconda3/envs/iris/lib/python3.8/site-packages/iris/coords.py", line 425, in __binary_operator__
    raise iris.exceptions.NotYetImplementedError(emsg)
iris.exceptions.NotYetImplementedError: Coord / AuxCoord
```
A workaround to avoid the error is that the user takes care of the order of "Cubes" and "Auxcoords". However, this can cause a problem in complicated equations including several terms. 

## How To Reproduce
Steps to reproduce the behaviour:

1. please see cube math documentation, section [Combining Multiple Phenomena to Form a New One](https://scitools-iris.readthedocs.io/en/v3.0.1/userguide/cube_maths.html#combining-multiple-phenomena-to-form-a-new-one).
2. first, load the pressure cube as it is explained.
3. then, define P0 as it is explained.
3. calculate the new pressure using the equation (pressure / p0): this works fine.
4. now make a different equation by changing the order i.e. (p0 / pressure): this returns the error, mentioned above.
5. replace the operator " / " with any operators of " + ", " - ", or " * " and repeat steps 3 and 4: the same things happen. 

## Environment 
 - OS & Version:  Ubuntu 20.04 LTS
 - Iris Version: 3.0.1
 - environment file is attached [iris.txt](https://github.com/SciTools/iris/files/5970971/iris.txt).

**Comment by rcomer** [comment]
An alternative workaround would be to define p0 as a scalar cube:

```python
p0 = iris.cube.Cube(1000.0, long_name='reference_pressure', units='hPa')
p0 / pressure
```
this has only really worked nicely since Iris 3.0.

If I keep p0 as an AuxCoord, I also get an error for `p0 / pressure` under Iris 2.4, so I'm curious about which Iris version ESMValTool used before.

**Comment by rcomer** [comment]
Ah, I see `Coord + Cube` and `Coord * Cube` _did_ work at Iris 2.4, and those seem to be the cases you are addressing at https://github.com/ESMValGroup/ESMValTool/pull/2023.

@bjlittle almost completely rewrote arithmetic for Iris 3.0 (making it better in all sorts of ways) so he'll know how easy it would be to reinstate that functionality.

**Comment by bjlittle** [comment]
@SarahAlidoost Apologies for the delay in getting back to you.

You've given me everything I need to replicate this issue and investigate further. So leave it with me and I'll take it from here 👍

**Comment by bjlittle** [comment]
Hey @SarahAlidoost,

Just to let you know that I can easily recreate this issue on Iris `master` and the release feature branch `v3.0.x`.

We're going to make an Iris `v3.0.2` patch release which will include the fix for this issue... it will also pull in at least one other bugfix in the patch release, which is another free win for the community :partying_face: 

You can track our progress with the `v3.0.2` release on this [Iris GitHub project](https://github.com/SciTools/iris/projects/30)... but I'll also update you here.

I don't have an ETA of when `v3.0.2` will be available, but I'm keen to resolve this ASAP. When tentative release dates do start to form, again I'll let you know here :+1:

**Comment by SarahAlidoost** [comment]
> Hey @SarahAlidoost,
> 
> Just to let you know that I can easily recreate this issue on Iris `master` and the release feature branch `v3.0.x`.
> 
> We're going to make an Iris `v3.0.2` patch release which will include the fix for this issue... it will also pull in at least one other bugfix in the patch release, which is another free win for the community 
> 
> You can track our progress with the `v3.0.2` release on this [Iris GitHub project](https://github.com/SciTools/iris/projects/30)... but I'll also update you here.
> 
> I don't have an ETA of when `v3.0.2` will be available, but I'm keen to resolve this ASAP. When tentative release dates do start to form, again I'll let you know here

thanks a lot for the explanations and update. Looking forward to v3.0.2 release :+1:

**Comment by bjlittle** [comment]
Hey @SarahAlidoost,

It's been a while, but I've not forgotten about you or this issue :smile: 

Just to let you know that we're in the final stages of pulling together `iris` v3.0.2 and I'm currently working on this issue right now.

I'm glad to say that I've managed to work out the root cause of the problem, and I'm now testing my fix. Once my fix is tested, reviewed and merged, we'll then release `iris` v3.0.2, which will be available before the end of next week i.e., before Friday 4th June 2021.

Thanks for your patience :+1:

**Comment by bjlittle** [comment]
Closed by #4159
